lib/spandex_tesla.ex

```diff
@@ -133,8 +142,20 @@ defmodule SpandexTesla do
     )
   end
 
-  defp span_result({:error, reason}, _measurements) do
-    tracer().span_error(%Error{message: inspect(reason)}, nil, [])
+  defp resource_name(metadata, config) do
+    get_resource_name = Keyword.get(config, :resource, &default_resource_name/1)
+
+    get_resource_name.(metadata)
+  end
+
+  defp default_resource_name(%{env: %{url: url, method: method}}) do
+    upcased_method = method |> to_string() |> String.upcase()
+    "#{upcased_method} #{url}"
+  end
+
+  defp default_resource_name(%{result: {:ok, %{method: method, url: url}}}) do
+    upcased_method = method |> to_string() |> String.upcase()
+    "#{upcased_method} #{url}"
   end
```

Comment:
```suggestion
  defp default_resource_name(%{env: %{url: url, method: method, opts: opts}}) do
    upcased_method = method |> to_string() |> String.upcase()
    resource_url = Keyword.get(opts, :req_url, url)

    "#{upcased_method} #{resource_url}"
  end

  defp default_resource_name(%{result: {:ok, %{method: method, url: url, opts: opts}}}) do
    upcased_method = method |> to_string() |> String.upcase()
    resource_url = Keyword.get(opts, :req_url, url)

    "#{upcased_method} #{resource_url}"
  end
```

@kzlsakal Maybe for a better behaviour we should support the req_url by default, using it if available. WDYT?

Reply:
That's great! Pushed the changes.